### ISSUE TITLE
Non-navigatable Grid has role=grid applied 

### ISSUE THREAD

**Issue by aleksandarevangelatov** [telerik/kendo-ui-core#7045]
### Bug report

When navigation for the Grid is disabled `navigatable: false` and there is no commands columns the table element of the Grid has `role=grid` applied. [The ARIA 1.2 definition of the role=grid](https://www.w3.org/TR/wai-aria-1.2/#grid) indicates the role should be applied to a composite widget containing a collection of one or more rows with one or more cells where some or all cells in the grid are focusable by using methods of two-dimensional navigation, such as directional arrow keys

### Reproduction of the problem
1. Run this [dojo](https://dojo.telerik.com/@AleksandarEvangelatov/UTUjIrir)
2. Inspect the `<table>` element of the Grid

### Expected/desired behavior
Refer to ticket 1577127 where the case is discussed.

### Environment

* **Kendo UI version:** 2022.3.914
* **Browser:** [all]

**Comment by aleksandarevangelatov** [comment]
@veselints @jivanova @dtopalov @ag-petrov @MilenaCh Can you share your feedback on the above and whether the behavior reported should be considered a Bug? I have discussed the item with @veselints and while in the example provided the `role=grid` may not conform the ARIA specification, there are scenarios where this might not be the case. For example, we provide the option for clients to set templates (i.e. arbitrary content, that might be focusable) for columns where the role seems to be valid as per the ARIA specification.

**Comment by veselints** [comment]
I have carefully reviewed the case again. In my opinion, we should take into account that the `role=grid` is applicable when

_"...some or all cells in the grid are focusable by using methods of two-dimensional navigation, such as directional arrow keys"._

Hence, when Grid is `navigatable: false`, even if having focusable elements in the cells (buttons in a command column, or focusable elements in a column template), its cells are not focusable and there is no arrow navigation among the cells. Having that said, I think that the `role=grid` should be removed in that scenario.

**Comment by ag-petrov** [comment]
Indeed when the grid is not navigatable we can safely remove the role in my opinion.
